src/Exports/DonorsExport.php

```diff
@@ -0,0 +1,199 @@
+
+    /** @inheritdoc */
+    protected $posted_data;
+
+    /** @var DateTime */
+    protected $startDate;
+
+    /** @var DateTime */
+    protected $endDate;
+
+    /**
+     * @inheritdoc
+     */
+    public function set_properties( $posted_data ) {
+        $this->posted_data = $posted_data;
+        ray( $this->posted_data );
+
+        if( ! $this->posted_data['start_date'] ) {
+            $this->startDate = date('Y-m-d', strtotime($this->posted_data['start_date']));
+        }
+
+        if( $this->posted_data['end_date'] ) {
+            $this->endDate = date('Y-m-d', strtotime($this->posted_data['end_date']));
+        }
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function csv_cols() {
+        return $this->flattenAddressColumn(
+            $this->filterIncludedColumns([
+                'full_name' => __( 'Name', 'give' ),
+                'email' => __( 'Email', 'give' ),
+                'address' => [
+                    'address_line1'      => __( 'Address', 'give' ),
+                    'address_line2'      => __( 'Address 2', 'give' ),
+                    'address_city'       => __( 'City', 'give' ),
+                    'address_state'      => __( 'State', 'give' ),
+                    'address_zip'        => __( 'Zip', 'give' ),
+                    'address_country'    => __( 'Country', 'give' ),
+                ],
+                'userid' => __( 'User ID', 'give' ),
+                'donations' => __( 'Number of donations', 'give' ),
+                'donation_sum' => __( 'Total Donated', 'give' ),
+        ]));
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function get_data()
+    {
+        $donorQuery = DB::table('give_donors', 'donors')
+            ->distinct()
+            ->select('donors.*');
+
+        $donationQuery = DB::table('posts', 'donations')
+            ->select('donations.ID', [ 'meta.meta_value', 'donorId'])
+            ->join(function(JoinQueryBuilder $builder) {
+                $builder
+                    ->leftJoin('give_donationmeta', 'meta')
+                    ->on('donations.ID', 'meta.donation_id')
+                    ->andOn('meta.meta_key', '_give_payment_donor_id', true);
+            })
+            ->where('donations.post_type', 'give_payment');
+
+        /*
+         * @TODO Toggle filter by donor registration date and donor's donation dates.
+         */
+
+        if( $this->startDate && $this->endDate ) {
+            $donationQuery->whereBetween('donations.post_date', $this->startDate, $this->endDate );
+        } elseif( $this->startDate ) {
+            $donationQuery->where('donations.post_date', $this->startDate, '>=');
+        } elseif( $this->endDate ) {
+            $donationQuery->where('donations.post_date', $this->endDate, '<');
+        }
+
+        $donorQuery->joinRaw( "JOIN ({$donationQuery->getSQL()}) AS sub ON donors.id = sub.donorId" );
+
+        $results = DB::get_results($donorQuery->getSQL(), ARRAY_A );
```

Comment:
@kjohnson you don't have to call `DB::get_results` to execute the query and get results. This functionality is already included in QB. You can do this instead `$results = $donorQuery->getAll(ARRAY_A);`